src/framework/theme/components/stepper/stepper.component.ts

```diff
@@ -121,12 +126,14 @@ export class NbStepperComponent {
   set selectedIndex(index: number) {
     if (!this.steps) {
       this._selectedIndex = index;
+      this.stepChanged.next(this._selectedIndex);
```

Comment:
We shouldn't emit change when having no steps as nothing is changed. Also, it will emit for initial value which is not desired behavior. Please, remove this line, if you don't have objections.